mindocr/data/transforms/abinet_transforms.py

```diff
@@ -0,0 +1,289 @@
+"""
+transform for text recognition tasks.
+"""
+
+import copy
+import random
+import re
+import warnings
+
+import cv2
+import numpy as np
+import PIL
+import six
+from PIL import Image
+
+import mindspore as ms
+import mindspore.dataset as ds
+
+from .svtr_transform import (
+    CVColorJitter,
+    CVGaussianNoise,
+    CVMotionBlur,
+    CVRandomAffine,
+    CVRandomPerspective,
+    CVRandomRotation,
+    CVRescale,
+)
+
+__all__ = ["ABINetTransforms"]
+
+
+class ABINetTransforms(object):
+    """Convert text label (str) to a sequence of character indices according to the char dictionary
+
+    Args:
+
+    """
+
+    def __init__(
+        self,
+        # start_token='<BOS>',
+        # end_token='<EOS>',
+        # unkown_token='',
+    ):
+        # ABINet_Transforms
+        self.case_sensitive = False
+        self.charset = CharsetMapper(max_length=26)
+
+    def __call__(self, data: dict):
+        img_lmdb = data["img_lmdb"]
+        label = data["label"]
+        label = label.encode("utf-8")
+        label = str(label, "utf-8")
+        try:
+            label = re.sub("[^0-9a-zA-Z]+", "", label)
+            if len(label) > 25 or len(label) <= 0:
+                string_false2 = f"en(label) > 25 or len(label) <= 0:   {label}, {len(label)}"
+                print(string_false2)
+            label = label[:25]
+            buf = six.BytesIO()
+            buf.write(img_lmdb)
+            buf.seek(0)
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore", UserWarning)
+                image = PIL.Image.open(buf).convert("RGB")
+            if not _check_image(image, pixels=6):
+                string_false1 = f"_check_image false:   {label}, {len(label)}"
+                print(string_false1)
+        except Exception:
+            string_false = f"Corrupted image is found:   {label}, {len(label)}"
+            print(string_false)
+
+        image = np.array(image)
+
+        text = label
+
+        length = len(text) + 1
+        length = float(length)
+
+        label = self.charset.get_labels(text, case_sensitive=self.case_sensitive)
+        label_for_mask = copy.deepcopy(label)
+        label_for_mask[int(length - 1)] = 1
+
+        label = onehot(label, self.charset.num_classes)
+        normalize_op = ms.dataset.vision.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225], is_hwc=False)
```

Comment:
Please decouple your augmentations as much as possible (move `Normalize` from `ABINetTransforms` to the config file).

Reply:
Modified, thank you.